apps/meteor/app/lib/server/functions/deleteRoom.ts

```diff
@@ -1,11 +1,12 @@
-import { Messages, Rooms, Subscriptions } from '@rocket.chat/models';
+import { Messages, Rooms, Subscriptions, ReadReceipts } from '@rocket.chat/models';
 
 import { callbacks } from '../../../../lib/callbacks';
 import { FileUpload } from '../../../file-upload/server';
 
 export const deleteRoom = async function (rid: string): Promise<void> {
 	await FileUpload.removeFilesByRoomId(rid);
 	await Messages.removeByRoomId(rid);
+	await ReadReceipts.removeByRoomId(rid);
```

Comment:
@sampaiodiego , wdyt of making this a `Promise.all` or `allSettled`?

Reply:
this whole PR was actually in my queue to take a deeper look, since I think this is a lot of changes whereas I think to trigger an action after a message was deleted should be only a matter of adding a hook (either a `this.onEvent` or `callback.add(...)`) somewhere.. 

maybe we don't have such hook, but we should.. imagine if for everything we want to do we need to find all these places again and add an additional call to do whatever we need. 🙈

Reply:
looking at this file specifically I can see there is a callback `afterDeleteRoom` .. so instead of adding the `ReadReceipts.removeByRoomId` here, we should actually call it from a callback for `afterDeleteRoom`.

Reply:
Moved it to the callback, tkss

I though about moving some more stuff into callbacks, but I guess that's not always possible with the callbacks we have by now.
1. For `cleanRoomHistory` we gotta know exactly what are the messages erased
2. For one specific message being deleted, I thought about calling this model from the `afterDeleteMessage` callback, but I'd have to replicate a big part of the code from the deleteMessage function (since I guess we'd only like to delete read receipts when `Message_KeepHistory` and `Message_ShowDeletedStatus` are disabled). Not sure what you think would be the best option in this case